llvm-10.0.1/llvm-crash-analyzer/lib/Analysis/TaintAnalysis.cpp

```diff
@@ -343,6 +344,25 @@ crash_analyzer::TaintAnalysis::isTainted(TaintInfo &Op,
   return Empty_op;
 }
 
+// This function is called when we do not know the source and destination
+// operands of a MIR during taint analysis.
+// If any of the machine operands in the given MIR is present in the taint
+// list, conservatively terminate the analysis.
+// Return true if OK to continue Analysis.
+// Return false if Analysis needs to be terminated.
+bool crash_analyzer::TaintAnalysis::continueAnalysis(
+    const MachineInstr &MI, SmallVectorImpl<TaintInfo> &TL,
+    RegisterEquivalence &REAnalysis) {
+  for (unsigned i = 0; i < MI.getNumOperands(); i++) {
+    TaintInfo Ti;
+    Ti.Op = &MI.getOperand(i);
```

Comment:
Ideally, we should have support for all(/most) of the MIR. If we do not recognize an MIR (which is a number of them at this time), currently we just continue the analysis ignoring that MIR. Obviously, skipping the taint propagation for such MIR, especially if it involves a tainted operand can lead to false positives and/or not finding any blame at all. This patch is a small (/conservative) improvement to the current situation. If an unsupported MIR has a tainted reg as an operand, we now conservatively choose to abort the analysis. 

For the example you point out, since we don't have the reg+offset combination available, we can only check for $rax, $rbp in the taint list. If rbp-8 was in the taint list and this MIR, say, modified the same location, we will miss it. This is same as the current behavior. However, if rax was in taint list, then we will abort since we aren't sure if this MIR modified it.